Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -145,199 +354,94 @@ namespace AzToolsFramework
         }
 
         // if we managed to add the bookmark
-        WriteBookmarksSettingsRegistryToFile();
+        WriteViewBookmarksSettingsRegistryToFile(localBookmarksFileName);
 
         return success;
     }
 
     bool LocalViewBookmarkLoader::SaveLastKnownLocation(const ViewBookmark& bookmark)
     {
-        return SaveLocalBookmark(bookmark, ViewBookmarkType::LastKnownLocation);
-    }
-
-    bool LocalViewBookmarkLoader::ReadViewBookmarksFromSettingsRegistry()
-    {
-        struct ViewBookmarkVisitor : AZ::SettingsRegistryInterface::Visitor
+        const LocalViewBookmarkComponent* bookmarkComponent = FindOrCreateLocalViewBookmarkComponent();
+        if (!bookmarkComponent)
         {
-            ViewBookmarkVisitor()
-                : m_viewBookmarksKey{ ViewBookmarksRegistryPath } {};
-
-            AZ::SettingsRegistryInterface::VisitResponse Traverse(
-                AZStd::string_view path,
-                AZStd::string_view,
-                AZ::SettingsRegistryInterface::VisitAction action,
-                AZ::SettingsRegistryInterface::Type) override
-            {
-                if (action == AZ::SettingsRegistryInterface::VisitAction::Begin)
-                {
-                    // strip off the last json pointer key from the path and if it matches the view bookmark key then add an entry
-                    // to the view bookmark map
-                    AZStd::optional<AZStd::string_view> localBookmarksID = AZ::StringFunc::TokenizeLast(path, "/");
-                    if (path == m_viewBookmarksKey && localBookmarksID && !localBookmarksID->empty())
-                    {
-                        AZStd::string_view bookmarkKey = localBookmarksID.value();
-
-                        if (auto existingBookmarkEntry = m_bookmarkMap.find(bookmarkKey) == m_bookmarkMap.end())
-                        {
-                            m_bookmarkMap.insert(AZStd::make_pair(bookmarkKey, AZStd::vector<ViewBookmark>{}));
-                        }
-                    }
-                }
+            AZ_Assert(bookmarkComponent, "Couldn't find or create LocalViewBookmarkComponent.");
```

Comment:
This should be an `AZ_Error` instead because it's handled. Otherwise if it's a coder error and just should never happen then remove the `if` and early return and keep the assert.

Reply:
Valid point will update 👍